### PR TITLE
ESP32: CAN module

### LINKED ISSUE #1948: ESP32: CAN module
### Missing feature
ESP32 has a built-in CAN controller. We need a module for it in nodemcu ~

### Justification
Why not? :)

### Workarounds
MCP2515 + SPI ? 

---------
https://esp32.com/viewtopic.php?f=12&t=380 Discussion
https://github.com/eerimoq/simba/commit/46778af636a4fe58773753b5ed9da83916cfaa06 Driver
https://github.com/ThomasBarth/ESP32-CAN-Driver/ ESP-IDF based driver

### PR DESCRIPTION
**Pull request by zelll**
Fixes #1948 .
- [x] This PR is for the `dev` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/en/*`.

ESP32 has a built-in CAN controller. With a CAN transceiver (like MCP1551/TJA1050), ESP32 can communicate with other nodes on a CAN bus, for example, OBD on a car. This module can read/send standard frame and extended frame. I have tested with a CAN-bus analyzer. 
This PR includes a driver based on https://github.com/ThomasBarth/ESP32-CAN-Driver/ . Espressif does not have a plan to add it (nor their own driver) to IDF.

### CODE REVIEW

File: docs/en/modules/can.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#### Syntax
+`can.stop()`
+
+#### Parameters
+
+#### Returns
+nil
+
+
+## can.send()
```
Comment (devsaurus) [inline docs/en/modules/can.md]: Cosmetic: Functions should be in alphabetical order.

File: components/modules/Kconfig [hunk 2]
```diff
@@ -67,6 +67,12 @@ config LUA_MODULE_BTHCI
   help
       Includes the simple BlueTooth HCI module.
 
+config LUA_MODULE_CAN
```
Comment (devsaurus) [inline components/modules/Kconfig]: I suppose that the `can` module requires the ESPCAN node. If so, then please establish the related dependencies between this item and the tree in components/driver_can/Kconfig. Something like "depends on" or "selects" from https://github.com/espressif/esp-idf/blob/master/tools/kconfig/kconfig-language.txt#L85.

Comment (zelll) [inline components/modules/Kconfig]: ESPCAN node was from original driver, but not necessary for nodemcu. I removed components/driver_can/Kconfig.

File: docs/en/modules/can.md (hunk 1 above)
Comment (zelll) [inline docs/en/modules/can.md]: moved `send()` to top

**Comment by devsaurus** [comment]
@zelll I updated to latest ESP-IDF which threw compile-time assertions for line [163 and 164](https://github.com/nodemcu/nodemcu-firmware/blob/dev-esp32/components/driver_can/CAN.c#L163). Had to change `(SET|CLEAR)_PERI_REG_MASK` to `DPORT_(SET|CLEAR)_PERI_REG_MASK`and hope the can module didn't break.

**Comment by zelll** [comment]
@devsaurus Tested. CAN works.

**Pull request merged**